picking.py

```diff
@@ -0,0 +1,167 @@
+
+    @classmethod
+    def get_to_pick(cls):
+        pool = Pool()
+        Shipment = pool.get('stock.shipment.out')
+        return [(s.id, s.rec_name) for s in Shipment.search([
+            # ('state', '=', 'assigned'),
+            ('state', '=', 'waiting'),
+            ])]
+
+
+class StockPickingShipmentOutScan(ModelView):
+    'Stock Picking Shipment Out Scan'
+    __name__ = 'stock.picking.shipment.out.scan'
+    shipment = fields.Many2One('stock.shipment.out', 'Shipment', readonly=True)
+    # pending_moves = fields.One2Many('stock.move', None, 'Pending Moves', readonly=True)
+    product = fields.Many2One('product.product', 'Product', readonly=True)
+    to_pick = fields.Char('To pick')
+    pending_moves = fields.Text('APP Pending Moves', readonly=True)
+
+
+class StockPickingShipmentOutResult(ModelView):
+    "Stock Picking Shipment Out Result"
+    __name__ = 'stock.picking.shipment.out.result'
+    shipment = fields.Many2One('stock.shipment.out', 'Shipment', readonly=True)
+    note = fields.Text('Note', readonly=True)
+
+
+class StockPickingShipmentOut(Wizard):
+    "Stock Picking Shipment Out Ask"
+    __name__ = 'stock.picking.shipment.out'
+    start_state = 'ask'
+    ask = StateView('stock.picking.shipment.out.ask',
+        'stock_scanner.stock_picking_shipment_out_start', [
+            Button('Cancel', 'end', 'tryton-cancel'),
+            Button('Scan', 'scan', 'tryton-ok', True),
+            ])
+    scan = StateView('stock.picking.shipment.out.scan',
+        'stock_scanner.stock_picking_shipment_out_scan', [
+            Button('Cancel', 'end', 'tryton-cancel'),
+            Button('Back', 'ask', 'tryton-back'),
+            Button('Pick', 'pick', 'tryton-launch', True),
+            Button('Packed', 'packed', 'tryton-ok'),
+            ])
+    pick = StateTransition()
+    packed = StateTransition()
+    result = StateView('stock.picking.shipment.out.result',
+        'stock_scanner.stock_picking_shipment_out_result', [
+            Button('Start', 'ask', 'tryton-back', True),
+            Button('Done', 'end', 'tryton-ok'),
+            ])
+
+    def transition_pick(self):
+        pool = Pool()
+        Shipment = pool.get('stock.shipment.out')
+
+        shipment = Shipment(self.scan.shipment)
+
+        def qty(value):
+            try:
+                return float(value)
+            except ValueError:
+                return False
+
+        shipment = Shipment(self.scan.shipment)
+        to_pick = self.scan.to_pick
+        quantity = qty(to_pick)
+
+        if shipment.scanned_product and quantity and len(to_pick) < 5:
+            shipment.scanned_quantity = shipment.scanned_uom.round(quantity)
+            shipment.save()
+            Shipment.scan([shipment])
+            shipment = Shipment(shipment.id)
+        else:
+            for move in shipment.pending_moves:
+                # TODO found product by party_code?
+                if move.product.party_code == to_pick:
```

Comment:
The module does not have a party_code field.

There should be a simple method to override the way to match the stock move line.

Probably something in the line:

```
    for move in shipment.pending_moves:
        if move.matches_scan(to_pick):
```

And then:

```
class Move:
    def matches_scan(self, input_):
        if move.product.code == input_:
            return True
        for identifier in move.product.identifiers:
            if identifier.code == input_:
                return True
        return False
```